src/main/java/seedu/address/ui/TaskCard.java

```diff
@@ -44,20 +51,38 @@
     private Label dependency;
     @FXML
     private FlowPane tags;
-    //TODO:Cannot find local variable tag
 
     public TaskCard(Task task, int displayedIndex) {
         super(FXML);
         this.task = task;
         id.setText(displayedIndex + ". ");
         name.setText(task.getName().fullName);
-        phone.setText(task.getDueDate().value);
-        address.setText(task.getDescription().value);
-        email.setText(task.getPriorityValue().value);
+        dueDate.setText(task.getDueDate().value);
+        remainingTime.setText(getRemainingTime());
+        description.setText(task.getDescription().value);
+        priorityValue.setText(task.getPriorityValue().value);
         status.setText(task.getStatus().toString());
-        hash.setText("id: " + Integer.toString(task.hashCode()));
-        dependency.setText("dependencies: " + task.getDependency().toString());
+        hash.setText(getHashId());
+        dependency.setText(getDependencies());
         task.getLabels().forEach(tag -> tags.getChildren().add(new Label(tag.labelName)));
+        registerAsAnEventHandler(this);
+    }
+
+    private String getHashId() {
+        return "id: " + Integer.toString(task.hashCode());
+    }
+
+    private String getDependencies() {
+        return "dependencies: " + task.getDependency().toString();
+    }
+
+    private String getRemainingTime() {
+        return "Remaining time: " + task.getTimeToDueDate();
```

Comment:
this is minor. id and dependencies are lower case but Remaining is upper?

Reply:
Resolved in latest commit